torch/csrc/jit/codegen/cuda/scheduler/registry.cpp

```diff
@@ -264,6 +267,81 @@ class SchedulerTopologyChecker {
 };
 } // namespace
 
+SchedulerRuntimeInfo::SchedulerRuntimeInfo(
+    Fusion* complete_fusion,
+    const at::ArrayRef<IValue>& inputs,
+    bool create_expr_evaluator)
+    : complete_fusion_(complete_fusion) {
+  collectAlignmentInfo(inputs);
+  if (create_expr_evaluator) {
+    initializeExpressionEvaluator(inputs);
+  }
+}
+
+SchedulerRuntimeInfo::SchedulerRuntimeInfo(
+    const SchedulerRuntimeInfo& copy_from)
+    : complete_fusion_(copy_from.complete_fusion_),
+      alignment_map_(copy_from.alignment_map_),
+      common_alignment_size_(copy_from.common_alignment_size_) {
+  expression_evaluator_ =
+      std::make_unique<ExpressionEvaluator>(complete_fusion_);
+}
+
+size_t SchedulerRuntimeInfo::getAlignmentSize(TensorView* tv) {
+  auto alignment_entry = alignment_map_.find(tv);
+  if (alignment_entry == alignment_map_.end()) {
+    return max_alignment_size_in_byte;
+  } else {
+    return alignment_entry->second;
+  }
+}
+
+void SchedulerRuntimeInfo::initializeExpressionEvaluator(
+    const at::ArrayRef<IValue>& inputs) {
+  expression_evaluator_ =
```

Comment:
nitpick: Looks like we could use an API clean-up on `bindFusionInputs`.
Can we put a TODO here?

Reply:
Sure. Could you give more details on how we'd like to refactor `bindFusionInputs`? I will either draft the comment or maybe just do it in this PR.

Reply:
It just looks somewhat awkward that we construct `expression_evaluator_` with `complete_fusion_` and then immediately rewrite it with the output from `bindFusionInputs`.

Might be cleaner to have it return a unique_ptr directly and adjust the other two use cases.

Reply:
I see. I agree it isn't great but I'd need to re-visit other use cases of `bindFusionInputs` to avoid too much changes on other logic. Added a todo and will cleanup in a follow up.  Thanks for pointing this out.